src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests.ActionCountTests.cs

```diff
@@ -0,0 +1,294 @@
+
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+namespace [||]NS1
+{{   
+    class Class1
+    {{
+    }}
+}}
+
+namespace NS2
+{{    
+    class Class1
+    {{
+    }}
+}}
+        </Document>
+    </Project>
+</Workspace>";
+
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task NoAction_MembersInBothGlobalAndNamespaceDeclaration_CursorOnNamespace()
+        {
+            var folders = new[] { "A", "B" };
+            var documentPath = CreateDocumentFilePath(folders);
+
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+namespace [||]NS1
+{{   
+    class Class1
+    {{
+    }}
+}} 
+
+class Class2
+{{
+}}
+        </Document>
+    </Project>
+</Workspace>";
+
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task NoAction_MembersInBothGlobalAndNamespaceDeclaration_CursorOnFirstGlobalMember()
+        {
+            var folders = new[] { "A", "B" };
+            var documentPath = CreateDocumentFilePath(folders);
+
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+class [||]Class1
+{{
+}}
+
+namespace NS1
+{{   
+    class Class2
+    {{
+    }}
+}} 
+        </Document>
+    </Project>
+</Workspace>";
+
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task NoAction_NestedNamespaceDeclarations()
+        {
+            var folders = new[] { "A", "B" };
+            var documentPath = CreateDocumentFilePath(folders);
+
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+namespace [||]NS1
+{{   
+    namespace NS2
+    {{
+        class Class1
+        {{
+        }}
+    }}
+}}
+        </Document>
+    </Project>
+</Workspace>";
+
+            await TestMissingInRegularAndScriptAsync(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task NoAction_InvalidNamespaceIdentifier()
```

Comment:
You can possibly combine large number of tests into a single unit test with different inputs using the `Theory` and `InlineData` attributes. For example, see http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs,19.

Reply:
Still seeing lot of duplication in tests here - was this considered?

---
In reply to: [224225062](https://github.com/dotnet/roslyn/pull/30156#discussion_r224225062) [](ancestors = 224225062)

Reply:
I personally prefer individual tests.

Reply:
@mavasani@microsoft.com Could you take another look at this PR please?

---
In reply to: [226485395](https://github.com/dotnet/roslyn/pull/30156#discussion_r226485395) [](ancestors = 226485395)